core/src/main/scala/org/zalando/grafter/Visualize.scala

```diff
@@ -0,0 +1,111 @@
+package org.zalando.grafter
+
+import scala.util.matching.Regex
+import org.bitbucket.inkytonik.kiama.relation.{Relation, Tree}
+
+trait Visualize {
+
+  type HashCode = Int
+
+  /**
+   * Generate a representation of the graph induced by the root component
+   * in GraphViz DOT format (http://www.graphviz.org)
+   *
+   * A filter can be used to filter out components
+   */
+  def asDotString[T <: Product](root:       T,
+                                filter:     Product => Boolean = _ => true,
+                                nodeFormat: String = "[shape=box]"): String = {
+
+    val relation = Relation.fromOneStep[Product](root, p => Tree.treeChildren(p).filter(filter))
+    val nodes = (relation.domain ++ relation.range).map(p => Node(p)).distinct
+    val indexes: Map[HashCode, (Int, Int)] = indexByIdentityHashCode(nodes)
+
+    val indexedNodes = nodes.map(_.setIndexes(indexes))
+    val arcs = relation.graph.map { case (s, t) => (Node(s, indexes), Node(t, indexes)) }.distinct
```

Comment:
I think that's because of the way we collect the relations in a DAG
```
    A
 /     \
B     C
   \   /
     D
      |
     E 
```
2 different paths from `A` will collect `D -> E`.